packages/components/src/navigation/NavTabList.tsx

```diff
@@ -0,0 +1,455 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import clamp from 'lodash.clamp';
+import {
+  DragDropContext,
+  Droppable,
+  OnDragEndResponder,
+} from 'react-beautiful-dnd';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { vsChevronRight, vsChevronLeft } from '@deephaven/icons';
+import { useResizeObserver } from '@deephaven/react-hooks';
+import DragUtils from '../DragUtils';
+import Button from '../Button';
+import NavTab from './NavTab';
+import './NavTabList.scss';
+import { ResolvableContextAction } from '../context-actions';
+
+// mouse hold timeout to act as hold instead of click
+const CLICK_TIMEOUT = 500;
+
+// mouse hold acceleration
+const START_SPEED = 0.01;
+const ACCELERATION = 0.0005;
+
+export interface NavTabItem {
+  key: string;
+  title: string;
+  isClosable?: boolean;
+}
+
+type NavTabListProps<T extends NavTabItem = NavTabItem> = {
+  activeKey: string;
+  tabs: T[];
+  onSelect: (key: string) => void;
+  onClose?: (key: string) => void;
+  onReorder: (sourceIndex: number, destinationIndex: number) => void;
+  isReorderAllowed: boolean;
+
+  /**
+   * Context items to add to the tab in addition to the default items.
+   * The default items are Close, Close to the Right, and Close All.
+   * The default items have a group value of 20.
+   *
+   * @param tab The tab to make context items for
+   * @returns Additional context items for the tab
+   */
+  makeContextItems?: (tab: T) => ResolvableContextAction[];
```

Comment:
I'd rather name this `contextActionsForTab` or `makeContextActions`. Should also be `ResolvableContextAction | ResolvableContextAction[]`

Reply:
The (small) benefit to just returning the array is it simplifies the merge code within this component. It's an infrequently used component, and the only actual difference in someone writing this function in DHE is adding square brackets to their return